SamMou.Api/Startup.cs

```diff
@@ -45,20 +38,24 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddControllers();
             services.AddScoped<IWeatherForecastService, WeatherForecastService>();
+            services.AddScoped<QueryObject>();
+            services.AddScoped<MutationObject>();
+            services.AddScoped<GraphQLSchema>();
+            services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+            services.AddSingleton<IDocumentWriter, DocumentWriter>();
 
-            // GraphQL            
-            GraphTypeTypeRegistry.Register<WeatherForecast, SamMouGraph>();
-            EfGraphQLConventions.RegisterInContainer(services, (context) => (context as GraphQLSamMouDataContext)?.Context);
-            EfGraphQLConventions.RegisterConnectionTypesInContainer(services);
-            services.AddSingleton<IDocumentExecuter, EfDocumentExecuter>();
-            services.AddSingleton<GraphQLQuery>();
-            services.AddSingleton<GraphQLSchema>();
+            services.AddApplicationInsightsTelemetry();
 
-            // Import also all the other created Graphs
-            foreach (Type type in GetGraphQlTypes())
-            {
-                services.AddScoped(type);
-            }
+            // GraphQL            
+            services.AddGraphQL((options, provider) =>{
+            // Log errors
+            var logger = provider.GetRequiredService<ILogger<Startup>>();
+                        options.UnhandledExceptionDelegate = ctx =>
+                            logger.LogError("{Error} occurred", ctx.OriginalException.Message);
+                    })
+                .AddGraphTypes(typeof(GraphQLSchema))
+                .AddDataLoader()
+                .AddSystemTextJson();

```

Comment:
is this dependency injection?

Reply:
Actually, what is this? adding logger into graphql?

Reply:
yup it's DI, and yup adding logger to graphQL will help us log error related to graphQL